gnovm/pkg/gnolang/preprocess.go

```diff
@@ -81,27 +83,55 @@ func PredefineFileSet(store Store, pn *PackageNode, fset *FileSet) {
 					// skip declarations already predefined
 					// (e.g. through recursion for a
 					// dependent)
-				} else {
-					// recursively predefine dependencies.
-					d2, _ := predefineNow(store, fn, d)
-					fn.Decls[i] = d2
+					continue
 				}
+
+				// recursively predefine dependencies.
+				d2, _ := predefineNow(store, fn, d)
+				fn.Decls[i] = d2
 			}
 		}
 	}
 	// Finally, predefine other decls and
 	// preprocess ValueDecls..
 	for _, fn := range fset.Files {
-		for i := 0; i < len(fn.Decls); i++ {
+		lenDecls := len(fn.Decls)
+		for i := 0; i < lenDecls; i++ {
 			d := fn.Decls[i]
 			if d.GetAttribute(ATTR_PREDEFINED) == true {
 				// skip declarations already predefined (e.g.
 				// through recursion for a dependent)
-			} else {
-				// recursively predefine dependencies.
-				d2, _ := predefineNow(store, fn, d)
-				fn.Decls[i] = d2
+				continue
+			}
+
+			if vd, ok := d.(*ValueDecl); ok && len(vd.NameExprs) > 1 && len(vd.Values) == len(vd.NameExprs) {
```

Comment:
Ok, does this now correctly handle the case where the variable dependencies are split across multiple packages?

You can test how this works by adding a package in `gnovm/tests/files/extern`; then importing the package in a file like in `gnovm/tests/files/redeclaration8.gno`.

Reply:
The issue this PR is solving  is not about that. This is only for a single global var statement.